lockbox-ios/Store/ItemDetailStore.swift

```diff
@@ -12,48 +12,64 @@ class ItemDetailStore: BaseItemDetailStore {
 
     private var dataStore: DataStore
     private var sizeClassStore: SizeClassStore
+    private var lifecycleStore: LifecycleStore
     private var userDefaultStore: UserDefaultStore
 
-    private var _itemDetailDisplay = ReplaySubject<ItemDetailDisplayAction>.create(bufferSize: 1)
+    private var _passwordRevealed = BehaviorRelay<Bool>(value: false)
 
-    lazy private(set) var itemDetailDisplay: Driver<ItemDetailDisplayAction> = {
-        return self._itemDetailDisplay.asDriver(onErrorJustReturn: .togglePassword(displayed: false))
+    lazy private(set) var passwordRevealed: Driver<Bool> = {
+        return self._passwordRevealed.asDriver(onErrorJustReturn: false)
     }()
 
     // RootPresenter needs a synchronous way to find out if the detail screen has a login or not
-    private(set) var itemDetailHasId = false
+    lazy private(set) var itemDetailHasId: Bool = {
+        return self._itemDetailId.value != ""
+    }()
 
-    init(dispatcher: Dispatcher = Dispatcher.shared,
-         dataStore: DataStore = DataStore.shared,
-         sizeClassStore: SizeClassStore = SizeClassStore.shared,
-         userDefaultStore: UserDefaultStore = UserDefaultStore.shared) {
+    init(
+            dispatcher: Dispatcher = .shared,
+            dataStore: DataStore = .shared,
+            sizeClassStore: SizeClassStore = .shared,
+            lifecycleStore: LifecycleStore = .shared,
+            userDefaultStore: UserDefaultStore = .shared
+    ) {
         self.dataStore = dataStore
         self.sizeClassStore = sizeClassStore
+        self.lifecycleStore = lifecycleStore
         self.userDefaultStore = userDefaultStore
 
         super.init(dispatcher: dispatcher)
 
         self.dispatcher.register
-            .filterByType(class: ItemDetailDisplayAction.self)
-            .bind(to: self._itemDetailDisplay)
-            .disposed(by: self.disposeBag)
+                .filterByType(class: ItemDetailDisplayAction.self)
+                .map { action -> Bool in
+                    switch action {
+                    case .togglePassword(let displayed):
+                        return displayed
+                    }
+                }
```

Comment:
Does this `switch` need to have a default case? If this is all the cases, then consider using `if case let`.